### ISSUE TITLE
[ML] Error using long context with Question Answering 

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#97917]
### Elasticsearch Version

8.9.0

### Java Version

_bundled_

### OS Version

any

### Problem Description

If the context text supplied to the Question Answering task is longer than the model's `max_sequence_length` and `truncate` is set to `none` then inference on the model fails with the message:

```
status_exception: question answering result has invalid dimension, expected 2 found [6] 
```
<img width="1438" alt="Screenshot 2023-07-25 at 08 21 37" src="https://github.com/elastic/elasticsearch/assets/2353640/12d296f6-03b2-4cda-89ce-7399348c4104">

### Steps to Reproduce

Use long input text to the question answering task

### Work Aroundf
The fix is to truncate the context which unfortunately may remove the part of the document containing the answer. For reference the call is documented here

```
POST _ml/trained_models/distilbert-base-cased-distilled-squad/deployment/_infer
{
  "docs":[{ "text_field": "question answering context"}],
  "inference_config": {
    "question_answering": {
      "question": "The question to ask",
      "tokenization": {
        "bert": {
          "truncate": "second",
          "span": -1          
        }
      }
    }
  }
}
````

**Comment by geehaad** [comment]
Is this issue solved?

**Comment by davidkyle** [comment]
> Is this issue solved?

Yes it is fixed in the latest release 8.10.0 and in 8.9.1
